src/modules/dataman/dataman.cpp

```diff
@@ -53,113 +54,24 @@
 
 #include "dataman.h"
 
-#if defined(FLASH_BASED_DATAMAN)
-#include <nuttx/clock.h>
-#include <nuttx/progmem.h>
-#endif
+#include "dataman_internal.h"
 
 __BEGIN_DECLS
 __EXPORT int dataman_main(int argc, char *argv[]);
 __END_DECLS
 
 static constexpr int TASK_STACK_SIZE = 1220;
 
-/* Private File based Operations */
-static ssize_t _file_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const void *buf,
-			   size_t count);
-static ssize_t _file_read(dm_item_t item, unsigned index, void *buf, size_t count);
-static int  _file_clear(dm_item_t item);
-static int  _file_restart(dm_reset_reason reason);
-static int _file_initialize(unsigned max_offset);
-static void _file_shutdown();
-
-/* Private Ram based Operations */
-static ssize_t _ram_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const void *buf,
-			  size_t count);
-static ssize_t _ram_read(dm_item_t item, unsigned index, void *buf, size_t count);
-static int  _ram_clear(dm_item_t item);
-static int  _ram_restart(dm_reset_reason reason);
-static int _ram_initialize(unsigned max_offset);
-static void _ram_shutdown();
+extern dm_operations_t dm_file_operations;
+extern dm_operations_t dm_ram_operations;
```

Comment:
Can you move all shared declarations into a common header?

Reply:
These ones are a bit of a mix that don't need to be shared everywhere. The goal here is to do an incremental step of shuffling the pieces together so that a followup as time allows could do something like put them in a DatamanBackend hierarchy (not sure yet).